### PR TITLE
feat: clean-up resharding V2 part 2

### PR DESCRIPTION
**Pull request by Trisfald**
Clean up resharding V2 code, more specifically:

- `StateChangesForResharding`
- `ReshardingResult`
- `ApplyResultForResharding`
- `ShardUpdateReason::Resharding`

Still TODO:
- `StateSnapshotType::ForReshardingOnly`
- `InvalidSplitShardsIds`
- `ShardSyncStatus::{ReshardingScheduling, ReshardingApplying}` 
- All resharding metrics
- `State_sync::resharding_state_roots`
- Resharding handle
- Reassigning of outgoing receipt in `chain/src/store/mod.rs`
- Remove `DBCol::StateChangesForSplitStates`
- Pytest

### CODE REVIEW

File: core/primitives/src/types.rs [hunk 1]
```diff
@@ -191,7 +190,7 @@ pub enum StateChangeCause {
     /// after protocol upgrade
     Migration,
     /// State changes for building states for re-sharding
-    Resharding,
+    ReshardingV2,
```
Comment (shreyan-gupta) [inline core/primitives/src/types.rs:193]: nit: Can change to `_UnusedResharding`

Comment (shreyan-gupta) [reply core/primitives/src/types.rs:193]: Similarly in StateChangeCauseView

**Review by shreyan-gupta — APPROVED** [review]
✨

File: core/primitives/src/types.rs (hunk 1 above)
Comment (shreyan-gupta) [reply core/primitives/src/types.rs:193]: Ignore this comment, I see the enum being used somewhere.

File: tools/protocol-schema-check/res/protocol_schema.toml
Comment (Trisfald) [inline tools/protocol-schema-check/res/protocol_schema.toml:1]: Can anyone double check this is alright?

`StateChangeCause::Resharding` has been renamed to `ReshardingV2` and `StateChangesForResharding` got removed

Comment (shreyan-gupta) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: It should be fine to get rid of StateChangesForResharding. For context, this is data written to RocksDB with every block that is used during the catchup phase of Resharding V2. Along with this, we should also be able to deprecate `DBCol::StateChangesForSplitStates`.

`StateChangeCause::Resharding` can be renamed

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -3662,33 +3653,6 @@ impl Chain {
         let prev_hash = block.header().prev_hash();
         let shard_context = self.get_shard_context(me, block.header(), shard_id, mode)?;
 
-        // We can only perform resharding when states are ready, i.e., mode != ApplyChunksMode::NotCaughtUp
```
Comment (wacban) [inline chain/chain/src/chain.rs:3665]: omg I love it gone so much

File: chain/chain/src/update_shard.rs [hunk 3]
```diff
@@ -35,55 +31,31 @@ pub struct OldChunkResult {
...
-    pub(crate) shard_uid: ShardUId,
-    pub(crate) results: Vec<ApplyResultForResharding>,
 }
 
 /// Result for a shard update for a single block.
 #[derive(Debug)]
 pub enum ShardUpdateResult {
     NewChunk(NewChunkResult),
     OldChunk(OldChunkResult),
-    Resharding(ReshardingResult),
```
Comment (wacban) [inline chain/chain/src/update_shard.rs:55]: ❤️

File: tools/protocol-schema-check/res/protocol_schema.toml
Comment (wacban) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: It looks good but can you double check that we don't need a protocol upgrade / db upgrade for this change? (This is what the schema was meant to detect).

**Review by wacban — APPROVED** [review]
Thanks so much!

File: tools/protocol-schema-check/res/protocol_schema.toml
Comment (Trisfald) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: > It looks good but can you double check that we don't need a protocol upgrade / db upgrade for this change? (This is what the schema was meant to detect).

Is it enough to create a node with latest release , swap the binary and check that it runs fine?

Comment (wacban) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: I was thinking about reading the code and checking where the modified structures were used. Running a node is good but we have canaries doing that also.

Comment (wacban) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: cc @Longarithm Can you advise?

Comment (Trisfald) [reply tools/protocol-schema-check/res/protocol_schema.toml:1]: code-wise it looks ok to me and a mainnet node stays in sync

**Pull request merged**